jwst/pathloss/pathloss.py

```diff
@@ -19,6 +20,25 @@
 NIRSPEC_IFU_SLICES = np.arange(30)
 
 
+def set_nans_to_donotuse(input_model):
+    """Set all NaN values in the data that have an even value to
+    DO_NOT_USE.
+
+    Data are modified in-place.
+
+    Parameters
+    ----------
+    input_model : jwst.datamodel.JwstDataModel
+        The input_model to be corrected.
+    """
+
+    # check the dq flags have the correct value
+    sci_nans = np.where(np.isnan(input_model.data), True, False)
+    dq_even = np.where(input_model.dq % 2 == 0, True, False)
+    if np.size(dq_even) > 0:
+        input_model.dq[sci_nans & dq_even] = dqflags.pixel['DO_NOT_USE']
```

Comment:
First line has a bit of redundancy, and second line may as well reference the dqflags just to avoid the assumption that odd will always be the DNU flag. I think np.where() is a bit more readable, but it's also somewhat slow.

Need to avoid overwriting other flags in the last line.

Suggested change:
```
    sci_nans = np.isnan(input_model.data)
    not_donotuse = ~(input_model.dq & dqflags.pixel['DO_NOT_USE']).astype(bool)
    if np.sum(sci_nans & not_donotuse) > 0:
        input_model.dq[sci_nans & not_donotuse] = input_model.dq[sci_nans & not_donutuse] | dqflags.pixel['DO_NOT_USE']
```

However, in thinking about how this method is being called, I think it would make sense to add this as a class method of JWSTDataModel. We could then call it when/where needed as `model.set_nans_to_donotuse()`. @braingram Any thoughts on this?

Reply:
Good point on the MultiSlit, I should have considered that. It just feels strange to make a lib function that only ever takes a datamodel and just edits some attributes of it. The method could contain checks on attribute existence? The thing with MultiSlitModel/MultiSpecModels is that their slits/spec lists should be composed of datamodels themselves. We could optionally implement a method in those datamodel definitions to overwrite the base class method to loop over their lists?

Maybe this is overkill, and a lib function just makes more sense. I do think this belongs in a general location though, as I see this being called in other steps in the near future.

Reply:
Would a function that takes in just the `data` and `dq` work? Something like:
```python
def set_nans_to_donotuse(data, dq):
    dq[np.isnan(data)] |= dqflags.pixel['DO_NOT_USE']
```

Reply:
That would work and is remarkably compact! @penaguerrero thoughts on using this, and maybe shifting it to `lib/` for future use in other steps?